Issue/95 attack log format

phfatmonkey opened the pull request:
updated format of Attack Log Display

to test

- open the Introductory Scenario and load Phase 1 (or complete until Phase 1 has been completed)
- make sure the game is in play mode
- click on the attack log button to open the attack log
- notice the attack log item is flashing red
- close the attack log with the X button
- reopen the attack log and notice the attack log item is no longer flashing

I did not add a graphic to the Attack Log button that indicates there are new attacks. I was going to handle that on another issue, because I want to make sure I'm handling the events the same way as the view tabs. I'll consult with Eric about that.

phfatmonkey committed:
first pass on the attack log format

phfatmonkey committed:
Merge branch 'develop' into issue/95-attack-log-format

phfatmonkey committed:
more work on attack log display

phfatmonkey committed:
assigned public fields in attack log item; updated script to iterate through all the log messages

phfatmonkey committed:
updated height of attack log item

phfatmonkey committed:
Merge branch 'develop' into issue/95-attack-log-format

phfatmonkey committed:
added close button to view header; had close button in attack log display

phfatmonkey committed:
changed function name to better reflect what it's doing -- SetAsRead

CowboyCoderForever commented:
closes #95

CowboyCoderForever commented:
- [x] Refactor `AttackLogDisplay` to use `DynamicList` and `DynamicListItem` derivatives for the management of Attack Log items.
- [ ] Refactor `AttackLogDisplay` to listen to changes to the Attack Log Scriptable variable directly, instead of getting supplied a list of items in the ShowDialog() method. (subscribe to the C# ScriptableVariable `OnValueChanged` perhaps?).
- [x] Modify the Attack Log button to indicate if there are new attack logs since the last time the log was viewed.
- [x] Remove the old cruft gameObjects and code files that are not used.

phfatmonkey committed:
made Attack Log Item into Dynamic List Item

erheine committed:
Separated where the attack logs get added to the AttackLogList so the dialog can be shown/hidden separately.

phfatmonkey committed:
updated the close button to match new style; updated attack log item to handle longer messages

phfatmonkey committed:
formatted string to separate time and log message

phfatmonkey committed:
added comments to code

phfatmonkey committed:
Merge branch 'develop' into issue/95-attack-log-format

CowboyCoderForever committed:
- Apply some code style.
- Hooked up the Attack Log Button so that it gets rendered differently when new attack logs come in.
